In [18]:
import sys
sys.path.append('../scripts/')

In [19]:
from engine_pass import postgresql as settings
from session import get_engine_from_settings, get_session


In [20]:
print(settings)

{'pguser': 'postgres', 'pgpass': 'postgres', 'host': 'localhost', 'port': 5432, 'pgdb': 'currencydb'}


In [21]:
engine = get_engine_from_settings()

In [22]:
import pandas as pd
df = pd.read_sql_query('SELECT * FROM xauusd_daily', con=engine)
df['date'] = pd.to_datetime(df['date'])


In [23]:
df.head(100)

id       date    open    high     low  close  tick_volume  spread  \
0     1 2002-01-01  278.20  278.70  276.70  278.7        351.0     0.0   
1     2 2002-01-02  278.20  278.95  277.35  278.5        291.0     0.0   
2     3 2002-01-03  278.00  279.75  276.55  278.2        621.0     0.0   
3     4 2002-01-04  277.55  279.15  276.80  278.7        356.0     0.0   
4     5 2002-01-07  277.90  280.01  276.90  278.5        563.0     0.0   
..  ...        ...     ...     ...     ...    ...          ...     ...   
95   96 2002-05-14  310.00  310.35  306.45  307.4        521.0     0.0   
96   97 2002-05-15  306.90  308.80  305.95  308.8        381.0     0.0   
97   98 2002-05-16  308.30  310.15  308.10  309.6        281.0     0.0   
98   99 2002-05-17  309.40  311.65  308.75  310.8        441.0     0.0   
99  100 2002-05-20  310.10  316.15  309.75  315.7        721.0     0.0   

    real_volume  
0           0.0  
1           0.0  
2           0.0  
3           0.0  
4           0.0  
..          ...  
95          0.0  
96          0.0  
97          0.0  
98          0.0  
99          0.0  

[100 rows x 9 columns]

In [24]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5068 entries, 0 to 5067
Data columns (total 9 columns):
 #   Column       Non-Null Count  Dtype         
---  ------       --------------  -----         
 0   id           5068 non-null   int64         
 1   date         5068 non-null   datetime64[ns]
 2   open         5068 non-null   float64       
 3   high         5068 non-null   float64       
 4   low          5068 non-null   float64       
 5   close        5068 non-null   float64       
 6   tick_volume  5068 non-null   float64       
 7   spread       5068 non-null   float64       
 8   real_volume  5068 non-null   float64       
dtypes: datetime64[ns](1), float64(7), int64(1)
memory usage: 356.5 KB


In [25]:
import plotly.express as px
fig = px.scatter(df, x = 'date', y = 'close', title = 'Daily Close price of XAUUSD', width = 1200, height = 700, opacity = 0.5)
# Add a slider to the plot
fig.update_xaxes(rangeslider_visible=True)
fig.update_yaxes(rangemode = 'normal')
fig.show()
# ['linear', 'hv', 'vh', 'hvh', 'vhv']

In [26]:

# to extract null values in terms of percentage
def null_percentage(dataframe):
    """ 
        This function outputs total null values in percentage
    """
    null_percentage = (dataframe.isna().sum()/dataframe.shape[0]*100)
    return null_percentage

null_percentage(df)
    



id             0.0
date           0.0
open           0.0
high           0.0
low            0.0
close          0.0
tick_volume    0.0
spread         0.0
real_volume    0.0
dtype: float64

In [27]:
# check stationarity

# AD Fuller method

import statsmodels.tsa.stattools as sts


In [29]:
sts.adfuller(df.open)

(-0.7220068794052158,
 0.8409666189909404,
 17,
 5050,
 {'1%': -3.4316455697178436,
  '5%': -2.862112502967502,
  '10%': -2.5670747438094303},
 39654.095593258615)

The Dickey-Fuller test gives us a test statistic of -0.811. When we compare this to the critical values at 1%, 5%, and 10% from the DF table, we see that the test statistic is greater than all of the critical values. This implies that there is no evidence to support the data being stationary.

The second value, 0.81, represents the p-value associated with the t-statistic. The p-value indicates a probability of 81% that we will not reject the null hypothesis. This means that we cannot confirm that the data comes from a stationary process.

The third value, 17, represents the number of lagged differences used in the test. This value indicates that there is some autocorrelation present in the data, going back 17 periods.

The fourth value, 5050, expresses the number of values used in the regression, which, along with the lag value, adds up to the total dataset.

Finally, the last value, 39654, represents the Akaike Information Criterion (AIC) estimated for the regression, assuming that there is some autocorrelation present in the data.